pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -0,0 +1,1150 @@
+		Stdin:       &container.Stdin,
+		StdinOnce:   &container.StdinOnce,
+	}
+
+	memoryLimit := container.Resources.Limits.Memory().Value()
+	cpuRequest := container.Resources.Requests.Cpu()
+	cpuLimit := container.Resources.Limits.Cpu()
+	var cpuShares int64
+	if cpuRequest.IsZero() && !cpuLimit.IsZero() {
+		cpuShares = milliCPUToShares(cpuLimit.MilliValue())
+	} else {
+		// if cpuRequest.Amount is nil, then milliCPUToShares will return the minimal number
+		// of CPU shares.
+		cpuShares = milliCPUToShares(cpuRequest.MilliValue())
+	}
+	if cpuShares != 0 || memoryLimit != 0 {
+		linuxResource := &runtimeApi.LinuxContainerResources{}
+		if cpuShares != 0 {
+			linuxResource.CpuShares = &cpuShares
+		}
+		if memoryLimit != 0 {
+			linuxResource.MemoryLimitInBytes = &memoryLimit
+		}
+		config.Linux = &runtimeApi.LinuxContainerConfig{
+			Resources: linuxResource,
+		}
+	}
+
+	if container.SecurityContext != nil {
+		securityContext := container.SecurityContext
+		if securityContext.Privileged != nil {
+			config.Privileged = securityContext.Privileged
+		}
+
+		if securityContext.ReadOnlyRootFilesystem != nil {
+			config.ReadonlyRootfs = securityContext.ReadOnlyRootFilesystem
+		}
+
+		if securityContext.Capabilities != nil {
+			if config.Linux == nil {
+				config.Linux = &runtimeApi.LinuxContainerConfig{
+					Capabilities: &runtimeApi.Capability{
+						AddCapabilities:  make([]string, 0, len(securityContext.Capabilities.Add)),
+						DropCapabilities: make([]string, 0, len(securityContext.Capabilities.Drop)),
+					},
+				}
+			}
+
+			for index, value := range securityContext.Capabilities.Add {
+				config.Linux.Capabilities.AddCapabilities[index] = string(value)
+			}
+			for index, value := range securityContext.Capabilities.Drop {
+				config.Linux.Capabilities.DropCapabilities[index] = string(value)
+			}
+		}
+
+		if securityContext.SELinuxOptions != nil {
+			if config.Linux == nil {
+				config.Linux = &runtimeApi.LinuxContainerConfig{}
+			}
+			config.Linux.SelinuxOptions = &runtimeApi.SELinuxOption{
+				User:  &securityContext.SELinuxOptions.User,
+				Role:  &securityContext.SELinuxOptions.Role,
+				Type:  &securityContext.SELinuxOptions.Type,
+				Level: &securityContext.SELinuxOptions.Level,
+			}
+		}
+	}
+
+	envs := make([]*runtimeApi.KeyValue, len(opts.Envs))
+	for index, e := range opts.Envs {
+		envs[index] = &runtimeApi.KeyValue{
+			Key:   &e.Name,
+			Value: &e.Value,
+		}
+	}
+	config.Envs = envs
+
+	return config, nil
+}
+
+func (m *kubeGenericRuntimeManager) startContainer(podSandboxID string, container *api.Container, pod *api.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []api.Secret) (error, string) {
+	err, msg := m.imagePuller.PullImage(pod, container, pullSecrets)
+	if err != nil {
+		return err, msg
+	}
+
+	ref, err := kubecontainer.GenerateContainerRef(pod, container)
+	if err != nil {
+		glog.Errorf("Can't make a ref to pod %v, container %v: '%v'", pod.Name, container.Name, err)
```

Comment:
nit: again, instead of using `pod.Name`, if you have a `api.Pod` object, just call `format.Pod()` to print relevant information when logging.